mithril-core/src/stm.rs

```diff
@@ -569,67 +589,105 @@ impl<D: Clone + Digest> StmSig<D> {
     /// Convert an `StmSig` into bytes
     ///
     /// # Layout
-    /// * Party id
     /// * Stake
     /// * Number of valid indexes (as u64)
     /// * Indexes of the signature
     /// * Public Key
     /// * Signature
-    /// * Merkle Tree path
+    /// * Merkle index of the signer.
     pub fn to_bytes(&self) -> Vec<u8> {
         let mut output = Vec::new();
         output.extend_from_slice(&self.stake.to_be_bytes());
         output.extend_from_slice(&(self.indexes.len() as u64).to_be_bytes());
+
         for index in &self.indexes {
             output.extend_from_slice(&index.to_be_bytes());
         }
+
         output.extend_from_slice(&self.pk.to_bytes());
         output.extend_from_slice(&self.sigma.to_bytes());
-        output.extend_from_slice(&self.path.to_bytes());
+
+        output.extend_from_slice(&self.signer_index.as_ref().unwrap().to_be_bytes());
         output
     }
 
-    /// Extract an `StmSig` from a byte slice.
-    pub fn from_bytes(bytes: &[u8]) -> Result<StmSig<D>, StmSignatureError<D>> {
+    /// Extract a batch compatible `StmSig` from a byte slice.
+    pub fn from_bytes(bytes: &[u8]) -> Result<StmSig<D>, StmSignatureError<D>>
+    where
+        D: Default + FixedOutput,
+    {
         let mut u64_bytes = [0u8; 8];
+
         u64_bytes.copy_from_slice(&bytes[..8]);
         let stake = u64::from_be_bytes(u64_bytes);
+
         u64_bytes.copy_from_slice(&bytes[8..16]);
         let nr_indexes = u64::from_be_bytes(u64_bytes) as usize;
-        let mut index = Vec::new();
+
+        let mut indexes = Vec::new();
         for i in 0..nr_indexes {
             u64_bytes.copy_from_slice(&bytes[16 + i * 8..24 + i * 8]);
-            index.push(u64::from_be_bytes(u64_bytes));
+            indexes.push(u64::from_be_bytes(u64_bytes));
         }
-        let pk = StmVerificationKey::from_bytes(&bytes[16 + nr_indexes * 8..])?;
-        let sigma = Signature::from_bytes(&bytes[112 + nr_indexes * 8..])?;
-        let path = Path::from_bytes(&bytes[160 + nr_indexes * 8..])?;
+
+        let offset = 16 + nr_indexes * 8;
+        let pk = StmVerificationKey::from_bytes(&bytes[offset..offset + 96])?;
+        let sigma = Signature::from_bytes(&bytes[offset + 96..offset + 144])?;
+
+        u64_bytes.copy_from_slice(&bytes[offset + 144..]);
```

Comment:
For consistency, you could use 
```suggestion
        u64_bytes.copy_from_slice(&bytes[offset + 144..offset + 152]);
```